### ISSUE TITLE
edges?

### ISSUE THREAD

**Issue by 7ammer** [hailwood/adonis-graphql#1]
Hello,
Just wondering what you're referring to when you mention edges in your code.

Cheers

**Comment by hailwood** [comment]
Hey @7ammer,

I've gone for a standard offset based pagination but used the relay connections as inspiration in terms of how the connections look as I really like it and it gives you the freedom later to add addition information to the relationship (meta data from a join table for example) without having to mess with your structure later.

Specifically a query over a connection (relationship) e.g. let's say blog posts have comments would look like this

```graphql
{
   blogPosts(id: 23) {
      id
      title
      contents
      comments(perPage: 10, page: 5) {
         pageInfo {
            hasNextPage
            totalEdges
         }
         edges {
            node {
               id
               title
               contents
            }
         }
   }
}
```

Basically it's just a pagination abstraction but you're free to either go 100% relay with cursors etc, or simplify it down to not have the concept of edges and nodes.

For reference here is the relay docs on connections.
https://facebook.github.io/relay/graphql/connections.htm

--
Matt

**Comment by 7ammer** [comment]
@hailwood
Ah makes sense now, cheers for the example. Guess I need to read up on a few more of these concepts!
